src/arizona_socket.erl

```diff
@@ -85,11 +79,11 @@ put_assign(Key, Value, Socket) ->
             }
     end.
 
--spec get_assigns(t()) -> assigns().
+-spec get_assigns(t()) -> arizona_template_renderer:assigns().
 get_assigns(#{assigns := Assigns}) ->
     Assigns.
 
--spec get_changes(t()) -> changes().
+-spec get_changes(t()) -> arizona_template_renderer:assigns().
```

Comment:
This is, at least, semantically confusing... changes and assigns are represented by the same thing?

Reply:
They are of the same type, [look](https://github.com/arizona-framework/arizona/blob/0f45120fccc8c00e1d687527889947806b21dcf9/src/arizona_socket.erl#L76).
"changes" are changes of the "assigns". It's just to track what changes will be applied as a patch at the DOM. After any "handle_event", "changes" is defined as an empty map [here](https://github.com/arizona-framework/arizona/blob/0f45120fccc8c00e1d687527889947806b21dcf9/src/arizona_websocket.erl#L94) in `arizona_socket:prune/1`.

Reply:
I understand. I'm just stating that they could've been defined with the same underlying type but different exported names, because semantically they'll mean different thigs...

Reply:
We can act differently on this. "changes" could be just a list of unique atoms representing what "keys" have changed. There is no need for it to be a map. The list of atoms is what the new [renderer needs](https://github.com/arizona-framework/arizona/pull/168/files#diff-f3d009d89c64dd10047d0941817744c4a47cd055cd94b9ae00a94912c197e30aR50), but It is called "ChangesVars" (typo here). Anyway, we must revisit this kind of stuff after the code is adapted for the new 
scanner, parser, compiler, and renderer. After this PR, I'll remove all the old modules. What do you think?

Reply:
Yeah, let's keep going forward.